### PR TITLE
Set the label text to be the children passed into the option component.

### PR DESCRIPTION
**Pull request by liondedan**
### Description
The <Option/> child property is set as the label text and the `value` is lifted to the <Select/> callback function. This allows for a different `value` and `title` to be so, so that when an option value is selected, the 'collapsed' text doesn't have to be the `value`.

### CODE REVIEW

File: src/molecules/Select/Option.tsx [hunk 1]
```diff
@@ -37,7 +37,7 @@ const Option: React.SFC<Props> = ({
   children,
   'data-qaid': qaId
 }) => {
-  const onChange = () => onClick(value);
+  const onChange = () => onClick(value, children);
```
Comment (prangarang) [inline src/molecules/Select/Option.tsx]: Where is the children prop defined in this?
```
        <Option key={index} value={e.title} onClick={onClick}>
          {e.title}
        </Option>
```

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@sofarsounds/maestro",
-  "version": "4.1.0",
+  "version": "5.1.0",
```
Comment (davidfloegel) [inline package.json]: `5.0.0` come on dan

File: src/molecules/Select/Option.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ interface Props {
   value?: any;
   disabled?: boolean;
   error?: boolean;
-  onClick: (value: any) => void;
+  onClick: (value: any, title: any) => void;
```
Comment (davidfloegel) [inline src/molecules/Select/Option.tsx]: is this any?

File: storybook/stories/Select.stories.tsx [hunk 4]
```diff
@@ -7,6 +7,11 @@ import { Select, Option } from '../../src';
 
 import cities from '../helpers/cities';
 
+const clickedSelect = (e: any) => {
+  action('Callback');
+  console.log(e);
```
Comment (davidfloegel) [inline storybook/stories/Select.stories.tsx]: are you joking dan!! console log???

File: storybook/stories/Select.stories.tsx [hunk 5]
```diff
@@ -15,7 +20,7 @@ storiesOf('Select', module)
 
       <Select
         readonly
-        handleOptionClick={action('Callback')}
+        handleOptionClick={e => clickedSelect(e)}
```
Comment (davidfloegel) [inline storybook/stories/Select.stories.tsx]: `handleOptionClick={clickSelect}`

File: src/molecules/Dropdown/__snapshots__/Dropdown.test.tsx.snap [hunk 6]
```diff
@@ -179,7 +179,7 @@ exports[`<Dropdown /> renders correctly 1`] = `
             "attrs": Array [],
             "componentStyle": ComponentStyle {
               "componentId": "sc-htpNat",
-              "isStatic": false,
```
[the remark this answers is not in the document]
Comment (liondedan) [reply src/molecules/Dropdown/__snapshots__/Dropdown.test.tsx.snap]: Something to do with us updating the styled-components package version - the only reason it's showing here is that it's running snapshot for the first time

File: src/molecules/Pagination/__snapshots__/Arrow.test.tsx.snap [hunk 7]
```diff
@@ -673,7 +673,7 @@ exports[`Pagination Arrow <ArrowLeft /> renders correctly 1`] = `
                 forwardedRef={null}
               >
                 <i
-                  className="c1 icon-chevronLeft "
+                  className="icon-chevronLeft  c1"
```
[the remark this answers is not in the document]
Comment (liondedan) [reply src/molecules/Pagination/__snapshots__/Arrow.test.tsx.snap]: Guessing it's the same as the above

File: src/molecules/Select/Option.tsx (hunk 3 above)
Comment (liondedan) [reply src/molecules/Select/Option.tsx]: Yes

File: storybook/stories/Select.stories.tsx (hunk 5 above)
Comment (liondedan) [reply storybook/stories/Select.stories.tsx]: The rule is you're never allowed to add this as a comment to one of my PRs

**Comment by davidfloegel** [comment]
# GitHub Pages Deployment
Pull Request:
https://sofarsounds.github.io/maestro/pull/198
Branch:
https://sofarsounds.github.io/maestro/branches/select-value-exposure
Storybook:
https://sofarsounds.github.io/maestro/4caab8e46eb6a24fa800dc0e65058ddb751f469c/
Build:
https://sofarsounds.github.io/maestro/4caab8e46eb6a24fa800dc0e65058ddb751f469c/dist.zip

**Pull request merged**